core/modules/comment/comment.install

```diff
@@ -363,6 +395,19 @@ function comment_update_1003() {
   }
 }
 
+/**
+ * Add an Override column for automatic comment closing for individual nodes.
+ */
+function comment_update_1004() {
+  if (db_field_exists('node', 'comment_close_override')) {
+    return;
+  }
+
+  $schema = array();
+  comment_schema_alter($schema);
```

Comment:
I missed this in my first review. Referencing a module's own hook_schema() or hook_schema_alter() in an update hook is a no-no. Because if hook_schema() needs to change, it will represent the most current version of the schema, but subsequent update hooks after this (say comment_update_1005()) would be working on the assumption that they are updating from the previous version.

Short-version is: we have to copy the default schema out of hook_schema_alter() into this update hook.